api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirRelatedPersonDaoImpl.java

```diff
@@ -9,14 +9,48 @@
  */
 package org.openmrs.module.fhir2.api.dao.impl;
 
+import java.util.Collection;
+
+import ca.uhn.fhir.rest.api.SortSpec;
+import ca.uhn.fhir.rest.param.DateRangeParam;
+import ca.uhn.fhir.rest.param.StringAndListParam;
+import ca.uhn.fhir.rest.param.TokenAndListParam;
 import lombok.AccessLevel;
 import lombok.Setter;
+import org.hibernate.Criteria;
 import org.openmrs.Relationship;
 import org.openmrs.module.fhir2.api.dao.FhirRelatedPersonDao;
 import org.springframework.stereotype.Component;
 
 @Component
 @Setter(AccessLevel.PACKAGE)
-public class FhirRelatedPersonDaoImpl extends BaseFhirDao<Relationship> implements FhirRelatedPersonDao {
+public class FhirRelatedPersonDaoImpl extends BaseRelatedPersonDao<Relationship> implements FhirRelatedPersonDao {
+	
+	private boolean isInnerQuery = true;
+	
+	@Override
+	public Collection<Relationship> searchRelationships(StringAndListParam name, TokenAndListParam gender,
+	        DateRangeParam birthDate, StringAndListParam city, StringAndListParam state, StringAndListParam postalCode,
+	        StringAndListParam country, SortSpec sort) {
+		
+		Criteria criteria = getSessionFactory().getCurrentSession().createCriteria(Relationship.class, "r");
+		
+		criteria.createAlias("r.personA", "m"); //inner join with person table
+		handleNames(criteria, name, null, null, isInnerQuery);
+		handleGender("m.gender", gender).ifPresent(criteria::add);
+		handleDateRange("m.birthdate", birthDate).ifPresent(criteria::add);
+		handlePersonAddress("pad", city, state, postalCode, country).ifPresent(c -> {
+			criteria.createAlias("m.addresses", "pad");
+			criteria.add(c);
+		});
+		
+		handleSort(criteria, sort);
+		
+		return criteria.list();
+	}
 	
+	@Override
+	protected String getSqlAlias() {
+		return "this_";
+	}
```

Comment:
I don't think we actually need this unless we're actually reusing the `BasePersonDao` class.

Reply:
Basically, this can be hard-coded where needed

Reply:
true thanks I fixed the mistake